modules/tracer-otel/src/main/java/org/opensearch/tracing/DefaultTracer.java

```diff
@@ -0,0 +1,238 @@
+ * It also handles tracing context propagation between spans.
+ *
+ * It internally uses OpenTelemetry tracer.
+ *
+ */
+public class DefaultTracer implements Tracer {
+
+    private static final Logger logger = LogManager.getLogger(DefaultTracer.class);
+    private static final String TRACE_ID = "trace_id";
+    private static final String SPAN_ID = "span_id";
+    private static final String SPAN_NAME = "span_name";
+    private static final String PARENT_SPAN_ID = "p_span_id";
+    private static final String THREAD_NAME = "th_name";
+    private static final String PARENT_SPAN_NAME = "p_span_name";
+    private static final String ROOT_SPAN = "RootSpan";
+
+    private final ThreadPool threadPool;
+    private final TracerSettings tracerSettings;
+    private final io.opentelemetry.api.trace.Tracer otelTracer;
+    private final OpenTelemetry openTelemetry;
+
+    /**
+     * Creates DefaultTracer instance
+     *
+     * @param openTelemetry Otel global Opentelemetry instance
+     * @param threadPool Thread pool
+     * @param tracerSettings tracer related settings
+     */
+    public DefaultTracer(OpenTelemetry openTelemetry, ThreadPool threadPool, TracerSettings tracerSettings) {
+        this.openTelemetry = openTelemetry;
+        this.otelTracer = openTelemetry.getTracer("os-tracer");
+        this.threadPool = threadPool;
+        this.tracerSettings = tracerSettings;
+    }
+
+    @Override
+    public void startSpan(String spanName, Level level) {
+        Span span = createSpan(spanName, getCurrentSpan(), level);
+        setCurrentSpanInContext(span);
+        setSpanAttributes(span);
+    }
+
+    @Override
+    public void endSpan() {
+        Span currentSpan = getCurrentSpan();
+        if (currentSpan != null) {
+            endSpan(currentSpan);
+            setCurrentSpanInContext(currentSpan.getParentSpan());
+        }
+    }
+
+    @Override
+    public void addAttribute(String key, String value) {
+        addSingleAttribute(AttributeKey.stringKey(key), value);
+    }
+
+    @Override
+    public void addAttribute(String key, long value) {
+        addSingleAttribute(AttributeKey.longKey(key), value);
+
+    }
+
+    @Override
+    public void addAttribute(String key, double value) {
+        addSingleAttribute(AttributeKey.doubleKey(key), value);
+    }
+
+    @Override
+    public void addAttribute(String key, boolean value) {
+        addSingleAttribute(AttributeKey.booleanKey(key), value);
+    }
+
+    @Override
+    public void addEvent(String event) {
+        Span currentSpan = getCurrentSpan();
+        if (currentSpan instanceof DefaultSpan && ((DefaultSpan) currentSpan).getOtelSpan() != null) {
+            ((DefaultSpan) currentSpan).getOtelSpan().addEvent(event);
+        }
+    }
+
+    @Override
+    public void close() {
+        if (openTelemetry instanceof Closeable) {
+            try {
+                ((Closeable) openTelemetry).close();
```

Comment:
The close is called only from Node.java when we want to close all the resources(including Otel). Also forcing closure on span could result incorrect data (wrong end time), so it might be ok to loose them instead of capturing incorrect spans